### PR TITLE
feat: support flownode

### PR DESCRIPTION
**Pull request by zyy17**
## What's changed

1. Add the new node role `flownode` in `GreptimeDBCluster`;
2. Improve e2e testing performance by remove the `LoadBalancer` setting;

### CODE REVIEW

**Comment by zyy17** [comment]
When https://github.com/GreptimeTeam/greptimedb/pull/4323 is merged, the PR will remove the setting of `frontend_addr` of flownode.

File: cmd/initializer/internal/config_generator.go [hunk 1]
```diff
@@ -34,7 +34,11 @@ type Options struct {
 	Namespace      string
 	ComponentKind  string
 
-	// For generating config of datanode.
+	// For generating config of datanode or flownode.
+	RPCPort     int32
```
Comment (daviderli614) [inline cmd/initializer/internal/config_generator.go:38]: question: Seem naming `FlownodeRPCPort` and `FlownodeServiceName` better ?

File: cmd/initializer/internal/config_generator.go [hunk 2]
```diff
@@ -123,31 +132,74 @@ func (c *ConfigGenerator) generateDatanodeConfig(initConfig []byte) ([]byte, err
...
-func (c *ConfigGenerator) allocateDatanodeID() (uint64, error) {
+func (c *ConfigGenerator) allocateNodeID() (uint64, error) {
 	name, err := c.hostname()
 	if err != nil {
 		return 0, err
 	}
 
 	if len(name) == 0 {
-		return 0, fmt.Errorf("the hostname is empty")
+		return 0, fmt.Errorf("the datanodeHostname is empty")
```
Comment (daviderli614) [inline cmd/initializer/internal/config_generator.go:188]: question: The allocateNodeID function contains flownode and datanode, use hostname replace datanodeHostname in error logging ?

File: tests/e2e/utils/utils.go [hunk 3]
```diff
@@ -206,3 +209,161 @@ func RunSQLTest(ctx context.Context, frontendIngressIP string, isDistributed boo
...
+
+	for i := 0; i < 20; i++ {
+		_, err = conn.ExecContext(ctx, insertDataSQL, i, timestampInMillisecond())
+		if err != nil {
+			return err
+		}
+	}
+
+	// Wait for the flow to process the data.
+	time.Sleep(5 * time.Second)
```
Comment (daviderli614) [inline tests/e2e/utils/utils.go:271]: Why sleep 5 second ?

File: cmd/initializer/internal/config_generator.go (hunk 2 above)
Comment (zyy17) [reply cmd/initializer/internal/config_generator.go:188]: typo. let me fix it.

File: cmd/initializer/internal/config_generator.go (hunk 1 above)
Comment (zyy17) [reply cmd/initializer/internal/config_generator.go:38]: I don't think so. The `DatanodeRPCPort` and `DatanodeServiceName` will be removed in the future. No need to name a `Flownode*`.

File: tests/e2e/utils/utils.go (hunk 3 above)
Comment (zyy17) [reply tests/e2e/utils/utils.go:271]: There is a small time window before the flownode receives the mirror requests and processes the data. Adding the sleeping logic will make sure we can always read the data.

**Review by daviderli614 — APPROVED** [review]
LGTM

**Pull request merged**